unit-tests/dds/test-librs-formats-converison.py

```diff
@@ -0,0 +1,178 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#test:donotrun:!dds
+
+from rspy import log, test
+import pyrealsense2 as rs
+import dds
+
+rs.log_to_console( rs.log_severity.debug )
+log.nested = 'C  '
+
+context = rs.context( '{"dds-domain":123,"dds-participant-name":"test-formats-conversion"}' )
+only_sw_devices = int(rs.product_line.sw_only) | int(rs.product_line.any_intel)
+
+import os.path
+cwd = os.path.dirname(os.path.realpath(__file__))
+remote_script = os.path.join( cwd, 'formats-conversion-server.py' )
+with test.remote( remote_script, nested_indent="  S" ) as remote:
+    remote.wait_until_ready()
+
+    # Tests setup
+    remote.run( 'create_server()' )
+    n_devs = 0
+    for dev in dds.wait_for_devices( context, only_sw_devices ):
+        n_devs += 1
+    test.check_equal( n_devs, 1 )
+    sensors = {sensor.get_info( rs.camera_info.name ) : sensor for sensor in dev.query_sensors()}
+    #
+    #############################################################################################
+    #
+    with test.closure( "Test Y8 conversion", on_fail=test.RAISE ):
+        if test.check( 'Y8-sensor' in sensors ):
+            sensor = sensors.get('Y8-sensor')
+            profiles = sensor.get_stream_profiles()
+
+            test.check_equal( len( profiles ), 1 ) # Y8 stays Y8, for infrared stream indexed 1
+            test.check_equal( profiles[0].format(), rs.format.y8 )
+            test.check_equal( profiles[0].stream_index(), 1 )
```

Comment:
Some cameras don't have a left and right... should the stream index be 0 or 1?

Reply:
The current Y8 converter is registered for index 1 only.

Reply:
Not sure what that means... For the L515, for example, there is only one IR. My understanding is that the stream index should be 0 (and greater than zero only when more than one exist). Is this a change in behavior? What was the result of this test before this PR?

Reply:
OK, I will rephrase, the current **D400** Y8 converter is for index 1 only.
The test device publishes "D400" as it's product line, [here](https://github.com/IntelRealSense/librealsense/pull/11850/files/645cf35ef1f9bdb56e654bbe85fd54b3499991c4#diff-10dba9177d24e8c03e5a711f90409d61abe2d33686efade5ac211ff98bbe8083R17)

Reply:
Let's talk because I'm still unclear :)